scripts/gen_syscalls.py

```diff
@@ -147,7 +147,7 @@ def analyze_fn(match_group):
     flat_args = [sys_id, func_name] + flat_args
     argslist = ", ".join(flat_args)
 
-    invocation = "%s(%s);" % (macro, argslist)
+    invocation = "%s(%s)" % (macro, argslist)
```

Comment:
Please add to commit message explicit mention of what is done here. E.g.

> Also update scripts/gen_syscalls.py to not generate extra semicolons after K_SYSCALL_DECLARE* macros.


Again, if this was my project, I would request to split this commit in 2, both with focused commit messages, as it updates 2 widely different parts of the codebase.